docs/core/configuration.md

```diff
@@ -604,3 +604,54 @@ Here's a brief summary of the timeouts:
 - `timeout_commit` = how long a validator should wait after committing a block, before starting
   on the new height (this gives us a chance to receive some more precommits,
   even though we already have +2/3)
+
+### The effect of `timeout_propose` on the proposer selection process
+
+Here's an interesting question. What if the particular validator sets a very
+small `timeout_propose`?
+
+Imagine there are only two validators in your network: Alice and Bob. Bob sets
+`timeout_propose` to 1s. Alice uses the default value of 3s. Bob will create
+blocks ~ every second, Alice - every 3 seconds (given `create_empty_blocks` is
+`true`). Let's say they both have an equal voting power. Given the proposer
+selection algorithm is a weighted round-robin, you may expect Alice and Bob to
+take turns proposing blocks, and the result will be:
+
+```
+#1 block - Alice
+#2 block - Bob
+#3 block - Alice
+#4 block - Bob
+...
+```
+
+What happens in reality is, however, a little bit different:
+
+```
+#1 block - Bob
+#2 block - Bob
+#3 block - Bob
+#4 block - Alice
+```
+
+That's because Bob is too fast at proposing blocks. This leaves Alice very
+little chances to propose a block and not always be catching up. Note every
+block Bob creates needs a vote from Alice to constitute 2/3+.
+
+Imagine now there are ten geographically distributed validators. One of them
+(Bob) sets `timeout_propose` to 1s. Now, Bob won't be able to move with the
```

Comment:
do we need to add that the other validators set the timeout_propose to a value of something like 3s or higher just to make it more clear?